File: android-client/app/src/main/java/com/componentvault/android/model/StockMovementRecord.kt
package com.componentvault.android.model

data class StockMovementRecord(
    val id: String,
    val componentId: String,
    val componentSku: String,
    val componentName: String,
    val movementType: String,
    val quantity: Int,
    val reason: String,
    val note: String,
    val happenedAt: String,
    val updatedAt: String,
    val deleted: Boolean,
    val locationId: String? = null,
    val destinationLocationId: String? = null,
) {
    /** Stored inbound/outbound quantities are magnitudes; adjustment quantities are signed. */
    val quantityChange: Long
        get() = when (movementType.lowercase(java.util.Locale.ROOT)) {
            "inbound" -> kotlin.math.abs(quantity.toLong())
            "outbound" -> -kotlin.math.abs(quantity.toLong())
            "transfer" -> 0L
            else -> quantity.toLong()
        }
}

File: windows-client/ComponentVault.WinUI/Models/StockMovementRecord.cs
namespace ComponentVault.WinUI.Models;

public sealed class StockMovementRecord
{
    public required string Id { get; init; }
    public required string ComponentId { get; init; }
    public required string ComponentSku { get; init; }
    public required string ComponentName { get; init; }
    public required string MovementType { get; init; }
    public required int Quantity { get; init; }
    public required string Reason { get; init; }
    public required string Note { get; init; }
    public required string HappenedAt { get; init; }
    public required string UpdatedAt { get; init; }
    public required bool Deleted { get; init; }

    public long QuantityChange => MovementType.Trim().ToLowerInvariant() switch
    {
        "outbound" => -Math.Abs((long)Quantity),
        "inbound" => Math.Abs((long)Quantity),
        _ => Quantity,
    };

    public string QuantityLabel => QuantityChange.ToString("+#,0;-#,0;0");

    public string MovementTypeLabel => MovementType.Trim().ToLowerInvariant() switch
    {
        "inbound" => "入库",
        "outbound" => "出库",
        "adjustment" => "调整",
        _ => MovementType,
    };
}
